crates/bevy_app/src/app.rs

```diff
@@ -765,9 +765,9 @@ impl App {
                 plugin_name: plugin.name().to_string(),
             })?;
         }
-        self.is_building_plugin = true;
+        self.building_plugin_counter += 1;
```

Comment:
Probably not something we need to fix in this PR, but this leaves `App` in an incorrect state if the plugin build panics. We may want to use `bevy_utils::CallOnDrop` here.

Reply:
`OnDrop` does not work here because we pass `self` to `Plugin::build`. I've used `catch_unwind` and `unwind_resume` instead. I am not sure if using `AssertUnwindSafe` here is incorrect, but I think this shouldn't be an issue because we are resuming the unwind before we can observe any broken invariants.